GitUI/CommandsDialogs/RevisionDiffControl.cs

```diff
@@ -1443,20 +1391,21 @@ private bool ResetSelectedFilesWithConfirmation()
             }
 
             // Reset to first (parent)
-            ResetSelectedItemsWithConfirmation(resetToSelected: false);
+            ResetSelectedItemsWithConfirmation(resetToParent: true);
             return true;
         }
 
-        private void ResetSelectedItemsWithConfirmation(bool resetToSelected)
+        private void ResetSelectedItemsWithConfirmation(bool resetToParent)
         {
             IEnumerable<FileStatusItem> items = DiffFiles.SelectedItems;
 
+            // new could be changed when restting, allow user to add to checkbox
```

Comment:
> You do not know if a file will be considered new after a checkout, like for renamed files.
This enables the checkbox, do not set it by default.

Does it cause an error message from git? If not and it only leaves a "new" file, the user can get rid of this "new" file by resetting again. Then the checkbox will be automatically ticked.

```suggestion
            // The "new" state could change when resetting, allow user to tick the checkbox.
```

Reply:
> Does it cause an error message from git? If not and it only leaves a "new" file, the user can get rid of this "new" file by resetting again. Then the checkbox will be automatically ticked.

The change is just about not disabling the checkbox. It is up to the user to tick the box or to redo the reset with leftover files. (There is no Git error message, the new status is detected by an additional git-status call.)

The handling was incomplete, for instance reset-to-selected were not considered at all.

The reverse approach to disable the checkbox when there are only modified files could work...

Can change the comment.

Reply:
> The reverse approach to disable the checkbox when there are only modified files could work...

Yes, please. I am used to the usually disabled checkbox. If it is enabled, I wonder why, and get distracted.